.github/workflows/upstream-dev-ci.yaml

```diff
@@ -0,0 +1,98 @@
+name: CI
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  schedule:
+    - cron: "0 0 * * *" # Daily “At 00:00” UTC
+  workflow_dispatch: # allows you to trigger the workflow run manually
+
+jobs:
+  upstream-dev:
+    name: upstream-dev
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.7", "3.8", "3.9"]
```

Comment:
the test matrix we have right now is `["3.6", "3.7", "3.8"]`. It is time to drop `3.6` and to start testing `3.9`, but maybe we should do that in a separate PR?

Reply:
I think we currently only do a `py38-upstream-dev` test. Not sure we actually need a matrix here.

Reply:
> I think we currently only do a py38-upstream-dev test. Not sure we actually need a matrix here.

If there is no need to test against multiple Python versions, I'm happy to remove the matrix, and stick with Python 3.8 only